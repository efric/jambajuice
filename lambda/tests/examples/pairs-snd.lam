(\ pair fst snd .

    snd (pair a b)

    -- Should reduce to b
)

-- pair
(\ x y f . f x y)

-- fst
(\ p . p (\ x y . x))

-- snd
(\ p . p (\ x y . y))
