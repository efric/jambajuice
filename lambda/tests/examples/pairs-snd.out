(\pair . \fst . \snd . snd (pair a b)) (\x . \y . \f . f x y) (\p . p (\x . \y . x)) (\p . p (\x . \y . y))
